meow/frontend/src/components/EditPost/index.js

```diff
@@ -149,11 +150,21 @@ class EditPost extends React.Component {
       });
   };
 
+  popSectionerror = () => {
+    notification.open({
+      message: "Send Failed :(",
+      description: "No section was selected",
+      icon: <Icon type="close-circle" style={{ color: "#FF0000" }} />
+    });
+  };
+
   sendNow = () => {
     const { postId } = this.props.match.params;
 
     this.savePostPromise(postId).then(data => {
-      if (data) {
+      ++this.state.version_number;
+      if (data && data.section !== null) {
+        this.setState({ sectionerror: "" });
```

Comment:
With this, if the state was originally 
```
{
   sections: ["arts", "opinion"],
   sectionerror: ""
}
```

And then you call `this.setState({ sectionerror: "" });`, then the state will become 

```
{
 sectionerror: ""
}
```

So you'll lose the `sections` part of the state. In order to perserve that, you can use the spread operator (`...this.state`). It's use is similar to the comment I made right above this comment.

Reply:
Ahh ok I was thinking about the setState we use when we are doing like functional components and it has a setState associated with each variable. Corrected!